### ISSUE TITLE
Types of property '__param_type__' are incompatible during next build

### ISSUE THREAD

**Issue by GorgeousPuree** [omermecitoglu/next-openapi-json-generator#21]
Console output during running next build:

PS C:\Users\Petya\WebstormProjects\aaa> npm run build

```
> aaa@0.1.0 build
> next build     

   ▲ Next.js 15.0.3
   - Environments: .env

   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types  ...Failed to compile.

.next/types/app/api/license/route.ts:49:7
Type error: Type '{ __tag__: "GET"; __param_position__: "second"; __param_type__: RouteHandlerContext<unknown> | undefined; }' does not satisfy the constraint 'ParamCheck<RouteContext>'.
  Types of property '__param_type__' are incompatible.
    Type 'RouteHandlerContext<unknown> | undefined' is not assignable to type 'RouteContext'.
      Type 'undefined' is not assignable to type 'RouteContext'.

  47 |     Diff<
  48 |       ParamCheck<RouteContext>,
> 49 |       {
     |       ^
  50 |         __tag__: 'GET'
  51 |         __param_position__: 'second'
  52 |         __param_type__: SecondArg<MaybeField<TEntry, 'GET'>>
```

license/route.ts:

```
import getLicenses from "@/operations/getLicenses";
import { LicenceDTO } from "@/models/license";
import defineRoute from "@omer-x/next-openapi-route-handler";
import z from "zod";

export const { GET } = defineRoute({
    operationId: "getLicenses",
    method: "GET",
    summary: "Get all users",
    description: "Retrieve a list of licenses",
    tags: ["Licenses"],
    queryParams: z.object({
      select: LicenceDTO.keyof().array().default([])
        .describe("List of the column names"),
    }),
    action: async () => {
      return Response.json(await getLicenses());
    },
    responses: {
      200: { description: "Returns a list of licenses", content: LicenceDTO, isArray: true },
    },
  });
```

This route works fine if I go to api/license, it returns correct information. However, during creation of optimized production build, I'm getting this error.

package.json:

```
{
  "name": "aaa",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@omer-x/next-openapi-json-generator": "1.3.0",
    "@omer-x/next-openapi-route-handler": "^1.6.0",
    "dotenv": "^16.4.5",
    "drizzle-orm": "^0.36.1",
    "drizzle-zod": "^0.5.1",
    "next": "15.0.3",
    "pg": "^8.13.1",
    "react": "19.0.0-rc-66855b96-20241106",
    "react-dom": "19.0.0-rc-66855b96-20241106",
    "swagger-ui-react": "^5.18.2",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@omer-x/openapi-types": "^1.1.0",
    "@types/node": "^20",
    "@types/pg": "^8.11.10",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "@types/swagger-ui-react": "^4.18.3",
    "drizzle-kit": "^0.28.0",
    "eslint": "^8",
    "eslint-config-next": "15.0.3",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "tsx": "^4.19.2",
    "typescript": "^5"
  }
}
```

**Comment by GorgeousPuree** [comment]
Seems to happen on Next 15. Build works fine on Next 14.
So I think this issue can be ignored for now.

**Comment by omermecitoglu** [comment]
Yeah, thanks for bringing this up. This is because of [Async Request APIs](https://nextjs.org/blog/next-15#async-request-apis-breaking-change)

**Comment by omermecitoglu** [comment]
Solved in version 1.8 of [the route handler module](https://github.com/omermecitoglu/next-openapi-route-handler/releases/tag/v1.8.0)
